### PR TITLE
Don't trigger reconnection before initialization complete (JAVA-954).

### PR DESCRIPTION
**Pull request by olim7t**

### CODE REVIEW

**Comment by adutra** [comment]
lgtm :+1:

File: driver-core/src/test/java/com/datastax/driver/core/ClusterInitTest.java [hunk 1]
```diff
@@ -144,6 +146,39 @@ public void should_handle_failing_or_missing_contact_points() throws UnknownHost
...
+     * @test_category connection
+     * @jira_ticket JAVA-954
+     * @expected_result No reconnection scheduled.
+     */
+    @Test(groups="short", expectedExceptions = NoHostAvailableException.class)
+    public void should_not_schedule_reconnections_before_init_complete() {
+        // Both contact points time out so we're sure we'll try both of them and init will never complete.
+        List<FakeHost> hosts = Lists.newArrayList(
+            new FakeHost(CCMBridge.ipOfNode(0), 9042, THROWING_CONNECT_TIMEOUTS),
+            new FakeHost(CCMBridge.ipOfNode(0), 9042, THROWING_CONNECT_TIMEOUTS));
```
Comment (tolbertam) [inline driver-core/src/test/java/com/datastax/driver/core/ClusterInitTest.java:161]: Should this be ipOfNode(1)?

**Comment by tolbertam** [comment]
looks good to me! :+1:

**Comment by olim7t** [comment]
Closing to reopen against 2.0 (#500).

**Pull request closed without merging**